test/e2e/persistent_volumes.go

```diff
@@ -41,110 +58,155 @@ func nfsServerPodCleanup(c *client.Client, config VolumeTestConfig) {
 	if config.serverImage != "" {
 		podName := config.prefix + "-server"
 		err := podClient.Delete(podName, nil)
-		if err != nil {
-			framework.Logf("Delete of %v pod failed: %v", podName, err)
-		}
+		Expect(err).NotTo(HaveOccurred())
 	}
 }
 
-// Delete the PV. Fail test if delete fails. If success the returned PV should
-// be nil, which prevents the AfterEach from attempting to delete it.
-func deletePersistentVolume(c *client.Client, pv *api.PersistentVolume) (*api.PersistentVolume, error) {
+// Cleanup up pvs and pvcs in multi-pv-pvc test cases. All entries found in the pv and
+// claims maps are deleted.
+// Note: this is the only code that deletes PV objects.
+func pvPvcCleanup(c *client.Client, ns string, pvols pvmap, claims pvcmap) {
+
+	if c != nil && len(ns) > 0 {
+		for pvcKey := range claims {
+			_, err := c.PersistentVolumeClaims(pvcKey.Namespace).Get(pvcKey.Name)
+			if !apierrs.IsNotFound(err) {
+				Expect(err).NotTo(HaveOccurred())
+				framework.Logf("   deleting PVC %v ...", pvcKey)
+				//err = c.PersistentVolumeClaims(pvcKey.Namespace).Delete(pvcKey.Name)
```

Comment:
Good catch! That line was commented as part of my testing to see what happens if Expect() is called inside an AfterEach. Thanks!